### PR TITLE
Table, Column, Property,Field_Properties: Added accessors.

### PR DESCRIPTION
**Pull request by judith-ipac**
add_to_property_tree(): Added check for duplicate 'arraysize' attribute. (Better fix to come.)

### CODE REVIEW

File: src/Table/generate_property_tree/add_to_property_tree.cxx [hunk 1]
```diff
@@ -35,18 +35,23 @@ void add_to_property_tree(const Column &column, const std::string &tree_name,
...
         // FIXME: This error is thrown a bit too late to be useful.
-
         if (a.first.empty())
             throw std::runtime_error("Empty attribute in field " + column.name +
                                      " which has type " + to_string(column.type));
+
+        if (added_arraysize && boost::equals(a.first, "arraysize")) {
+            continue;
+        }
+
```
Comment (smonkewitz) [inline src/Table/generate_property_tree/add_to_property_tree.cxx]: (This seems unrelated to accessors.) Do I understand correctly that user specified arraysizes will now always be overwritten with `*` for string columns? Why the change?

File: src/Field_Properties.hxx [hunk 2]
```diff
@@ -42,5 +42,31 @@ public:
...
+    }
+
+    inline void set_description(const std::string &desc) { description.assign(desc); }
+    inline void set_attributes(const std::map<std::string, std::string> &attrs) {
+        attributes = attrs;
+    }
+    inline void set_values(const Values &vals) { values = vals; }
+    inline void set_links(const std::vector<std::pair<std::string, std::string> > &ls) {
+        links = ls;
+    }
```
[the remark this answers is not in the document]
Comment (smonkewitz) [reply src/Field_Properties.hxx]: I think the same `constexpr` comment applies in the other files.

[the remark this answers is not in the document]
Comment (judith-ipac) [reply src/Field_Properties.hxx]: Thanks, will make the constexpr changes.

The nice thing about having public accessors and private members is that you can change the implementation and replace the private members with something altogether different and your clients won't even notice.

File: src/Table/generate_property_tree/add_to_property_tree.cxx (hunk 1 above)
Comment (judith-ipac) [reply src/Table/generate_property_tree/add_to_property_tree.cxx]: I thought of putting in a comment with a reference to whatever it was that made me do this, but it was so easy to find such references that I didn't bother.  Now of course I can't find any.  I don't remember whether it was an external test that was failing or a warning in an IVOA spec.  I'll see if I can find it and if I can't I'll back this out.

File: src/Field_Properties.hxx (hunk 2 above)
[the remark this answers is not in the document]
Comment (smonkewitz) [reply src/Field_Properties.hxx]: I tend to agree.

File: src/Table/generate_property_tree/add_to_property_tree.cxx (hunk 1 above)
Comment (smonkewitz) [reply src/Table/generate_property_tree/add_to_property_tree.cxx]: The VOTable spec says:
```
Strings, which are defined as a set of characters, can therefore be represented in VOTable as a fixed- or variable- length array of characters...
```
If `*`s are needed for some specific `<FIELD>`s, then maybe fixing the arraysize attributes for those specific fields would be better?

Comment (smonkewitz) [reply src/Table/generate_property_tree/add_to_property_tree.cxx]: On the other hand, tablator doesn't support unbounded length strings, and `*` doesn't seem like it is ever actually wrong, so I am OK with this change. (I'm still curious what exactly motivated this though.)

Comment (judith-ipac) [reply src/Table/generate_property_tree/add_to_property_tree.cxx]: Yes, I certainly wouldn't have dreamt it up myself so *something* must have triggered it.  I'll see if I can track down the reference after I make the other changes.  At the least I could add that attribute only when it isn't already present.

File: src/Field_Properties.hxx (hunk 2 above)
[the remark this answers is not in the document]
Comment (judith-ipac) [reply src/Field_Properties.hxx]: The compiler didn't like the way I tried to use constexpr.  I don't understand why it *should* work for getters and setters---isn't the idea that constexpr functions can be evaluated at compile time?

File: src/Table/generate_property_tree/add_to_property_tree.cxx (hunk 1 above)
Comment (judith-ipac) [reply src/Table/generate_property_tree/add_to_property_tree.cxx]: Ah, right.  These lines `  if (column.type == Data_Type::CHAR || column.array_size != 1)
    field.add ("<xmlattr>.arraysize", "*");`
have been there preceding the loop through attributes since before my time.  I just added a check during the loop so that we don't wind up with two "arraysize" attributes if the user happened to supply one.

File: src/Field_Properties.hxx (hunk 2 above)
[the remark this answers is not in the document]
Comment (smonkewitz) [reply src/Field_Properties.hxx]: I'm guessing this is because there is no constexpr constructor. But it isn't possible to provide one, because the member types don't have them. `const` it is (sorry for the detour).

File: src/Table/generate_property_tree/add_to_property_tree.cxx (hunk 1 above)
Comment (smonkewitz) [reply src/Table/generate_property_tree/add_to_property_tree.cxx]: I see - thanks for clarifying!
